db/postgres.go

```diff
@@ -0,0 +1,189 @@
+	}
+
+	sqlDb, err := p.getSqlConnection()
+	if err != nil {
+		return 0, fmt.Errorf("could not get applied migrations amount: %s", err)
+	}
+
+	appliedMigrations, err := migrate.GetMigrationRecords(sqlDb, "postgres")
+	if err != nil {
+		return 0, fmt.Errorf("could not get applied migrations amount: %s", err)
+	}
+
+	return len(appliedMigrations), nil
+}
+
+// getSqlConnection returns a generic SQL database interface using the GORM interface.
+func (p *Postgres) getSqlConnection() (*sql.DB, error) {
+	db, err := gorm.Open(postgres.Open(p.db.String()), &gorm.Config{Logger: logger.Default.LogMode(logger.Silent)})
+	if err != nil {
+		return nil, fmt.Errorf("could not get sql connection: %s", err)
+	}
+
+	sqlDb, err := db.DB()
+	if err != nil {
+		return nil, fmt.Errorf("could not get sql connection: %s", err)
+	}
+
+	return sqlDb, nil
+}
+
+func (p *Postgres) getMigrationsSource() *migrate.FileMigrationSource {
+	if p.db.MigrationsTable != "" {
+		migrate.SetTable(p.db.MigrationsTable)
+	}
+	return &migrate.FileMigrationSource{Dir: p.db.MigrationsPath}
+}
+
+// getPendingMigrationsCount returns the number of pending migrations to be applied.
+// If the function is unable to interact w/ the database,
+// 0 is returned with the error that occured in interacting with the database.
+func (p *Postgres) GetPendingMigrationsCount() (int, error) {
+	migrationsSource := p.getMigrationsSource()
+	sqlDb, err := p.getSqlConnection()
+	if err != nil {
+		return 0, fmt.Errorf("could not get pending migrations count: %s", err)
+	}
+
+	plannedMigrations, _, err := migrate.PlanMigration(sqlDb, "postgres", migrationsSource, migrate.Up, math.MaxInt32)
+	if err != nil {
+		return 0, fmt.Errorf("could not get pending migrations count: %s", err)
+	}
+
+	return len(plannedMigrations), nil
+}
+
+// CreateDatabaseIfNotExists checks that connecting to the provided Postgres database is possible.
+// If it does not exist, it will retry after creating the database if the create flag is set.
+// If the connection still fails or the create flag is not set, it will return an error.
+func (p *Postgres) CreateDatabaseIfNotExists() error {
+	doesExist, err := p.IsDatabaseCreated()
+	if err != nil {
+		return fmt.Errorf("failed to check if database exists: %s", err)
+	}
+	if doesExist {
+		return nil
+	}
+	fmt.Println("Database does not exist. Creating database...")
+	// Try to create database connecting to default postgres database on same host with same user
+	db_copy := *p.db
+	db_copy.DBName = "postgres"
+	pg_db, err := gorm.Open(postgres.Open(db_copy.String()), &gorm.Config{Logger: logger.Default.LogMode(logger.Silent)})
+	if err != nil {
+		return fmt.Errorf("could not connect to postgres database: %s", err)
+	}
+	stmt := fmt.Sprintf("CREATE DATABASE %s;", p.db.DBName)
+	if rs := pg_db.Exec(stmt); rs.Error != nil {
+		return fmt.Errorf("could not create database: %s", rs.Error)
+	}
+	return nil
+}
+
+// EraseDatabase drops the database if the drop flag is set.
```

Comment:
please update comment
```suggestion
// EraseDatabase is a destructive command that will drop configured database.
```